Makefile.am

```diff
@@ -1,5 +1,9 @@
 ACLOCAL_AMFLAGS = ${ACLOCAL_FLAGS} -I m4
 
+if P4THRIFT
+  AM_CPPFLAGS += -DP4THRIFT
```

Comment:
I don't think this propagate to the subdirs, have you checked? You may need to add it to `MY_CPPFLAGS` in `configure.ac`